test/test_other.py

```diff
@@ -16095,3 +16095,23 @@ def test_em_js_bool_macro_expansion(self):
 
   def test_getifaddrs(self):
     self.do_other_test('test_getifaddrs.c')
+
+  @parameterized({
+    'web': ('web',),
+    'node': ('node',),
+  })
+  def test_unsupported_min_version_when_unsupported_env(self, env):
+    create_file('pre.js', '''
+      #preprocess
+      var MIN_NODE_VERSION = {{{ MIN_NODE_VERSION }}};
+      var MIN_CHROME_VERSION = {{{ MIN_CHROME_VERSION }}};
+      var MIN_SAFARI_VERSION = {{{ MIN_SAFARI_VERSION }}};
+      var MIN_FIREFOX_VERSION = {{{ MIN_FIREFOX_VERSION }}};
```

Comment:
Can we write this test by putting this in the pre.js instead:

```
#if !ENVIRONMENT_MAY_BE_NODE
assert(MIN_NODE_VERSION == 0x7FFFFFFF)
#endif

#if !ENVIRONMENT_MAY_BE_WEB
assert(MIN_CHROME_VERSION == 0x7FFFFFFF)
assert(MIN_SAFARI_VERSION == 0x7FFFFFFF)
assert(MIN_FIREFOX_VERSION == 0x7FFFFFFF)
#endif
```

I don't love trying read/understand `assertContainedIf` type things.

Reply:
I tried something similar first, but unfortunately that doesn't work because when the test is built in `-sENVIRONMENT=web` mode, it complains about not being able to run in Node.js (which is our default test runner).

Reply:
If you have a better idea, I'm happy to switch though.

Reply:
> I tried something similar first, but unfortunately that doesn't work because when the test is built in `-sENVIRONMENT=web` mode, it complains about not being able to run in Node.js (which is our default test runner).

Ah yes of course.